### ISSUE TITLE
Cross-compiling problems

### ISSUE THREAD

**Issue by staticfloat** [ampl/mp#115]
Hello, I am trying to cross-compile `ASL` for a variety of platforms, and I would really like to avoid the bootstrapping issues of generating `arith.h` and `expr-info.cc`.  As a bit of background, I am trying to build redistributable binaries for the [Julia project](https://julialang.org), and our binary building infrastructure is based on cross-compilation.

Right now, when I try to cross-compile for Windows, it tries to use `wine` to run the newly created executables, but they look like they're just generating headers/source.  Is it possible for ASL to detect we're cross-compiling, and generate the necessary executables for the host build system and use those to generate the headers/source?  Or even more simply, is there a way I can manually construct these headers and just use those in my cross-compilation?

Thanks!

**Comment by staticfloat** [comment]
For the record, I am working around this right now by just [precomputing the set of `arith.h` files](https://github.com/staticfloat/mp-extra) that I need across all the platforms that I intend to build for and [jamming that into the build tree](https://github.com/staticfloat/IpoptBuilder/blob/0e3cfdf14ddfae017a03b5537ab12b5325d7757e/build_tarballs.jl#L42-L48) such that it doesn't _need_ to run `wine`.

**Comment by staticfloat** [comment]
True; it doesn't look like `expr-info.cc` changes across architectures though; I believe it should be possible to compile it with `$(HOSTCC)`, and use the output regardless of whether we're cross-compiling or not.

**Comment by zhihaoy** [comment]
This is necessary to get `arm64-windows` and `arm64-uwp` build in vcpkg. (`x64-uwp` can also benefit from this).

**Comment by fdabrandao** [comment]
A default `arith.h` file (https://github.com/ampl/mp/blob/master/src/asl/solvers/arith.h1) was added in d03e486f98ad45b7ec2898480f943e9c846f2df0 so that `arithchk` is not necessary anymore unless the compiler does not support something in the default `arith.h` file. If the `arith.h` file used by default does not match the target architecture, or if the compiler is not sufficiently compatible with gcc or Microsoft C/C++, run `cmake -DBUILD=all -DGENERATE_ARITH=true ..` to generate an architecture-specific `arith.h` file with `arithchk` (just as in the previous procedure).

**Comment by JackBoosY** [comment]
I can use non-arm executable to generate header files, but I encountered other errors:
```
WS2_32.Lib(WS2_32.dll) : fatal error LNK1112: module machine type 'x64' conflicts with target machine type 'ARM64'
```
